acnportal/acnsim/network/charging_network.py

```diff
@@ -8,9 +8,13 @@ class ChargingNetwork:
     """
     The ChargingNetwork class describes the infrastructure of the charging network with
     information about the types of the charging station_schedule.
+
+    Args:
+        violation_tolerance (float): Absolute amount by which an input
+            charging schedule may violate network constrants (A).
     """
 
-    def __init__(self):
+    def __init__(self, violation_tolerance=1e-5):
```

Comment:
I've set violation_tolerance as a constant here, meaning each constraint allows the same amount of violation. Should the violation_tolerance be more general, i.e. a vector giving a tolerance for each constraint individually?

Reply:
I think may make sense to have an absolute tolerance like we have now as well as a relative tolerance expressed as a percent violation.  Then we use the larger of the two tolerances.

Reply:
A vector of tolerances is more flexible but might be annoying to use, esspecially as the number of constraints increases.

Reply:
We'll do a vector of tolerances if the need arises; for now, we'll use a constant. I've pushed a version with relative tolerances.